bridge/contracts/BToken.sol

```diff
@@ -185,6 +192,62 @@ contract BToken is
         return numBTK;
     }
 
+    /// Burn the tokens without sending ether back to user as the normal burn
+    /// function. The generated ether will be distributed in the distribute method.
+    function destroyBTokens(uint256 numBTK_) public returns (bool) {
+        _destroyTokens(msg.sender, numBTK_);
+        return true;
+    }
+
+    /// Burn pre-approved tokens without sending ether back to an account as the
+    /// normal burn function. The generated ether will be distributed in the
+    /// distribute method. Mostly used to burn tokens in a third party contract
+    /// without having to transfer before.
+    function destroyPreApprovedBTokens(address account, uint256 numBTK_) public returns (bool) {
+        ERC20Upgradeable._spendAllowance(account, msg.sender, numBTK_);
+        _destroyTokens(account, numBTK_);
+        return true;
+    }
+
+    function depositTokensOnBridges(
```

Comment:
Three concerns:
First I would either like a lock to protect re-entry OR the docs must explain why re-entry is not a concern. Any explanation should be very detailed and any other functions that may change these concerns should carry information about what MUST not be changed.
Second I would like this logic to attempt to eat from existing btoken balance if eth is sent but not sufficient. 
Third maxTokens should be minTokens since we do not care if we get more but we do care if we do not get enough from perspective of deposit fees.